core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -149,13 +150,17 @@ public long length() {
     return writer.length();
   }
 
+  void useSequenceNumber(long sequenceNumber) {
```

Comment:
All of the other configuration is passed in when creating the writer. Is there a reason not to do that here?

Reply:
I think the cleanest way to do this is to add a new `add` method:

```java
  /**
   * Add an added entry for a file at a specific sequence number.
   * <p>
   * The entry's snapshot ID will be this manifest's snapshot ID.
   *
   * @param addedFile a data file
   * @param sequenceNumber sequence number for the data file
   */
  public void add(F addedFile, long sequenceNumber) {
    addEntry(reused.wrapAppend(snapshotId, sequenceNumber, addedFile));
  }
```

Reply:
sounds good, will do it through `add`